rfcs/0182-taskcluster-yml-remote-references.md

```diff
@@ -0,0 +1,49 @@
+# RFC 182 - Allow remote references to .taskcluster.yml files processed by Taskcluster-GitHub
+* Comments: [#182](https://github.com/taskcluster/taskcluster-rfcs/pull/182)
+* Proposed by: @bhearsum
+
+# Summary
+
+Allow `.taskcluster.yml` files in GitHub repositories to outsource most of their contents to a `.taskcluster.yml` stored elsewhere.
+
+## Motivation
+
+`.taskcluster.yml` files in GitHub repositories are responsible for creating tasks (or not) in response to various GitHub events. In some cases (eg: when a decision task is used to delegate most of this to a separate tool), these files end up looking almost entirely identical, and hundreds of lines of JSON-e ends up duplicated across many different repositories. This duplication is a significant maintenance burden, and easily results in unwanted differences.
+
+Allowing a `.taskcluster.yml` to reference a file stored elsewhere will allow for deduplication of the bulk of the content in these files, resolving the concerns noted above.
+
+# Details
+
+Two new variables will be introduced to `.taskcluster.yml` files. The first, `config-from`, will allow an URL to be provided that points at a full-fledged `.taskcluster.yml` hosted elsewhere. The second, `context`, will allow for variables to be defined that will be included with the JSON-e context when the referenced `.taskcluster.yml` is rendered. Here is an example of a possible concrete `.taskcluster.yml`:
```

Comment:
What if the base also has `config-from`? Will nested references be allowed? If so, is it worth checking for circular references?

Reply:
There's a note further down that nested references will explicitly _not_ be allowed. (It doesn't mean we can never support them...but I don't see a reason to add this complexity until we know there's a good reason to support it.)